processor/src/main/java/com/dropbox/kaiken/processor/InjectableActivityValidator.kt

```diff
@@ -1,55 +1,53 @@
-
-        val typeElement = annotatedActivity.annotatedActivityElement
-
-        return when {
-            !typeElement.isPublic() -> {
-                messager.error(typeElement, "The class ${typeElement.qualifiedName} is not public")
-                false
-            }
-
-            typeElement.isAbstract() -> {
-                messager.error(typeElement, "The class ${typeElement.qualifiedName} is abstract")
-                false
-            }
-
-            !typeElement.isAndroidActivity() -> {
-                messager.error(
-                    typeElement,
-                    "The class ${typeElement.qualifiedName} is not an Android activity"
-                )
-                messager.printMessage(Diagnostic.Kind.NOTE, typeElement.superclass.kind.name)
-                false
-            }
-
-            !typeElement.implementsInjectionHolder() -> {
-                messager.error(
-                    typeElement,
-                    "The class ${typeElement.qualifiedName} does not implement" +
-                        " DependencyProviderResolver"
-                )
-                typeElement.interfaces.forEach {
-                    messager.printMessage(Diagnostic.Kind.NOTE, it.kind.name)
-                }
-                false
-            }
-
-            else -> {
-                true
-            }
-        }
+import org.jetbrains.kotlin.descriptors.ClassDescriptor
+import org.jetbrains.kotlin.descriptors.ClassifierDescriptor
+import org.jetbrains.kotlin.descriptors.DescriptorVisibilities
+import org.jetbrains.kotlin.name.FqName
+import org.jetbrains.kotlin.psi.KtClassOrObject
+import org.jetbrains.kotlin.resolve.DescriptorUtils
+import org.jetbrains.kotlin.resolve.descriptorUtil.fqNameSafe
+import org.jetbrains.kotlin.resolve.descriptorUtil.getAllSuperClassifiers
+
+internal fun validateActivity(descriptor: ClassDescriptor?, clazz: KtClassOrObject) {
+    check(descriptor != null) {
+        "The ClassDescriptor for ${clazz.name} is null"
+    }
+
+    check(clazz.fqName != null) {
+        "The ClassDescriptor for ${clazz.name} is null"
+    }
+
+    val fqName: FqName = requireNotNull(clazz.fqName) {
+        "The fully qualified name for ${clazz.name} is null"
+    }
+
+    check(descriptor.visibility == DescriptorVisibilities.PUBLIC) {
+        "The class ${fqName.shortName()} is not public"
+    }
+
+    check(!DescriptorUtils.classCanHaveAbstractDeclaration(descriptor)) {
+        "The class ${fqName.shortName()} is abstract"
+    }
+
+    check(descriptor.isActivity()) {
+        "The class ${fqName.shortName()} is not an Android activity. Found: ${
+        descriptor.getAllSuperClassifiers().toList().map { it.name }
+        }"
+    }
+
+    check(descriptor.implementsInjectorHolder()) {
+        "The class ${fqName.shortName()} does not implement" +
+            " DependencyProviderResolver"
     }
 }
+
+internal fun ClassifierDescriptor.isActivity(): Boolean =
```

Comment:
Not call this isAndroidxActivty or actually check whether it's a component activity

Reply:
I'll use similar naming as before https://github.com/dropbox/kaiken/blob/9a10c03f71d82fb783995308e96d5a620dc17b45/processor/src/main/java/com/dropbox/kaiken/processor/internal/ElementUtils.kt#L34

Reply:
Hmm OK it was bad naming. Also I can't remember why we needed componentactivity but that was a recent change